### PR TITLE
[#115] Update sdk version to support priority

### LINKED ISSUE #115: Update neofs-sdk-go/pool for priority support
Implemented in https://github.com/nspcc-dev/neofs-sdk-go/issues/74

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #115

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,23 +69,24 @@ In general, everything available as CLI parameter can also be specified via
...
-$ HTTP_GW_PEERS_0_ADDRESS=192.168.130.72:8080 HTTP_GW_PEERS_0_WEIGHT=9 \
-  HTTP_GW_PEERS_1_ADDRESS=192.168.130.71:8080 HTTP_GW_PEERS_1_WEIGHT=1 neofs-http-gw
+$ HTTP_GW_PEERS_0_ADDRESS=192.168.130.71:8080 HTTP_GW_PEERS_0_WEIGHT=1 HTTP_GW_PEERS_0_PRIORITY=1 \
+  HTTP_GW_PEERS_1_ADDRESS=192.168.130.72:8080 HTTP_GW_PEERS_1_WEIGHT=9 HTTP_GW_PEERS_1_PRIORITY=2 \
+  HTTP_GW_PEERS_2_ADDRESS=192.168.130.73:8080 HTTP_GW_PEERS_2_WEIGHT=1 HTTP_GW_PEERS_2_PRIORITY=2 \
+  neofs-http-gw
 ```
-This command will make gateway use 192.168.130.72 for 90% of requests and
-192.168.130.71 for remaining 10%.
+This command will make gateway use 192.168.130.71 while it is healthy. Otherwise, it will make use 192.168.130.72
```
Comment (masterSplinter01) [inline README.md:88]: Maybe make the gateway use? `Otherwise, it will make use 192.168.130.72` sounds strange a bit.

File: README.md [hunk 2]
```diff
@@ -69,23 +69,24 @@ In general, everything available as CLI parameter can also be specified via
 environment variables, so they're not specifically mentioned in most cases
 (see `--help` also). If you prefer a config file you can use it in yaml format.
 
-### Nodes and weights
+### Nodes: weights and priority
```
Comment (532910) [inline README.md:72]: Here and elsewhere: should be "weights and priorities", shouldn't it?

File: README.md [hunk 3]
```diff
@@ -69,23 +69,24 @@ In general, everything available as CLI parameter can also be specified via
...
 You can specify multiple `-p` options to add more NeoFS nodes, this will make
-gateway spread requests equally among them (using weight 1 for every node):
+gateway spread requests equally among them (using weight 1 and priority 1 for every node):
 
 ```
 $ neofs-http-gw -p 192.168.130.72:8080 -p 192.168.130.71:8080
 ```
-If you want some specific load distribution proportions, use weights, but they
-can only be specified via environment variables:
+If you want some specific load distribution proportions, use weights and priority:
```
Comment (532910) [inline README.md:80]: priorities

File: uploader/upload.go [hunk 4]
```diff
@@ -195,9 +195,10 @@ func (pr *putResponse) encode(w io.Writer) error {
 func getEpochDurations(ctx context.Context, p pool.Pool) (*epochDurations, error) {
 	if conn, _, err := p.Connection(); err != nil {
 		return nil, err
-	} else if networkInfo, err := conn.NetworkInfo(ctx); err != nil {
+	} else if networkInfoRes, err := conn.NetworkInfo(ctx); err != nil {
```
Comment (alexvanin) [inline uploader/upload.go:198]: While https://github.com/nspcc-dev/neofs-sdk-go/issues/91 is not done, we have to use more complex error handling approach, like it is done in [pool](https://github.com/nspcc-dev/neofs-sdk-go/blob/a5b61c4f64779ff9ccd7dda39df0d6630d469a9d/pool/pool.go#L542-L546)

```go
	if conn, _, err := p.Connection(); err != nil {
		return nil, err
	} else if networkInfoRes, err := conn.NetworkInfo(ctx); err != nil {
		return nil, err
	} else if err := apistatus.ErrFromStatus(res.Status()); err != nil {
		return nil, err
	} else {
		...
```

**Comment by alexvanin** [comment]
I think test are failed, because clients with status codes do not work well with old storage nodes. We made compatibility of old client with new nodes, but new clients with old nodes are not going to work, unfortunately.

I think we can drop all previous aio-testcontainer versions and start from v0.27.0, it is available in [DockerHub](https://hub.docker.com/layers/nspccdev/neofs-aio-testcontainer/0.27.0/images/sha256-fb96cbf71f02f2ae2570729fa2ae5f2a713a5e575357ba1ed60b32ab16bfda49?context=repo)
/cc @realloc

**Comment by KirillovDenis** [comment]
It seems the reason of failing is trying to use `0.27.0` and some other version (even the same "latest").
So maybe we need to fix using multiple versions in tests or use just `latest`

**Comment by KirillovDenis** [comment]
> It seems the reason of failing is trying to use `0.27.0` and some other version (even the same `latest`). So maybe we need to fix using multiple versions in tests or use just `latest`

Or problem in latest `neofs-aio-testcontainer` image

File: uploader/upload.go [hunk 5]
```diff
@@ -9,6 +9,8 @@ import (
 	"strconv"
 	"time"
 
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+
```
Comment (alexvanin) [inline uploader/upload.go:13]: Empty line.

**Review by alexvanin — COMMENTED** [review]
The test is passed on my machine even with previous versions. But it doesn't work in GitHub actions :thinking:

**Comment by alexvanin** [comment]
According to logs, there was some kind of race or bad timing at container creation operation. With [1 second blocks](https://github.com/nspcc-dev/neofs-aio/pull/10) container approved almost immediately, so the problem should gone.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**